File: pptx-service/renderers/quote.py
from builder import add_bg, add_textbox, add_rect
from theme import ThemeColors


def render(slide, data: dict, t: ThemeColors) -> None:
    add_bg(slide, t.bg)
    quote = data.get("quote", "")
    source = data.get("source", "")

    add_textbox(slide, 0.7, 0.8, 2.0, 2.5, "“",
                t.font_display, 180, t.accent, bold=True)

    add_textbox(slide, 1.2, 1.5, 11.0, 4.0, quote,
                t.font_display, 44, t.text, italic=True, wrap=True)

    add_rect(slide, 1.2, 6.1, 1.5, 0.04, fill_hex=t.accent)
    add_textbox(slide, 1.2, 6.2, 8.0, 0.6, f"— {source}",
                t.font_body, 22, t.muted)

File: lib/pptx/slides/quote.ts
import type { QuoteSlide } from '@/lib/types';
import {
  py, SLIDE_W,
  setBackground, addTextBox, addPageNumber,
  type PptxTheme,
} from '../helpers';
import type PptxGenJS from 'pptxgenjs';
type PSlide = ReturnType<InstanceType<typeof PptxGenJS>['addSlide']>;

export function render(slide: PSlide, data: QuoteSlide, t: PptxTheme, n: number, total: number): void {
  setBackground(slide, '#' + t.bg);

  addTextBox(slide, '“', {
    x: t.paddingIn, y: py(80), w: py(160),
    fontSize: Math.round(t.heroSize * 1.5), color: '#' + t.accent, fontFace: t.fontDisplay, bold: true,
  });

  addTextBox(slide, data.quote, {
    x: t.paddingIn, y: py(200), w: SLIDE_W - 2 * t.paddingIn,
    fontSize: t.sectionSize, color: '#' + t.text, fontFace: t.fontDisplay, italic: true, align: 'center', wrap: true,
  });

  addTextBox(slide, `— ${data.source}`, {
    x: t.paddingIn, y: py(700), w: SLIDE_W - 2 * t.paddingIn,
    fontSize: t.captionSize, color: '#' + t.muted, fontFace: t.fontBody, align: 'center',
  });

  addPageNumber(slide, n, total, t);
}
